pkg/ctl/cmdutils/nodegroup_flags.go

```diff
@@ -39,7 +39,7 @@ func AddCommonCreateNodeGroupFlags(fs *pflag.FlagSet, cmd *Cmd, ng *api.NodeGrou
 	ng.SSH.PublicKeyPath = fs.String("ssh-public-key", "", "SSH public key to use for nodes (import from local path, or use existing EC2 key pair)")
 	ng.SSH.EnableSSM = fs.Bool("enable-ssm", false, "Enable AWS Systems Manager (SSM)")
 
-	fs.StringVar(&ng.AMI, "node-ami", "", "'auto-ssm', 'auto', 'static' (deprecated) or an AMI id (advanced use)")
+	fs.StringVar(&ng.AMI, "node-ami", "", "'auto-ssm', 'auto' or an AMI ID (advanced use)")
```

Comment:
could we set a default string? Instead of relying on empty string meaning the user didn't set it

Reply:
>could we set a default string? Instead of relying on empty string meaning the user didn't set it


If we set a default string, in addition to coming up with a name for the default behaviour, we would also have to validate against the user setting that default value. This isn't a behaviour we want to expose to end users. 

>Instead of relying on empty string meaning the user didn't set it

The empty string (or the zero value) is often used to signal default behaviour, and is often preferred because it doesn't require a type to be initialised with certain values for it to be usable. This also means that code that calls `ResolveAMI` elsewhere, including tests, would have to ensure `ng.AMI` is initialised with a default value.

Reply:
What value does `""` represent though? What does 
```
		resolver = ami.NewMultiResolver(
			ami.NewSSMResolver(provider.SSM()),
			ami.NewAutoResolver(provider.EC2()),
		)
```

Mean? If we have a value for this the user can be specific in wanting this and note that we default to it

Reply:
> The AMI resolution is an implementation detail that users should not have to care about. 

But we do expose it for them to configure? This combination is exposed to the user, the way they can configure it so by omitting instead of specifying, which is what I think I find weird.

> This is also not a feature we want exposed to end users, because once SSM support for Ubuntu is out, the AutoResolver will be deprecated. We'll then have to handle deprecation and removal of the feature.

Ok I'm happy for it to stay as is then :+1:

Reply:
>But we do expose it for them to configure? This combination is exposed to the user, the way they can configure it so by omitting instead of specifying, which is what I think I find weird.

Giving it a specific value that describes the behaviour and letting users configure it will expose it to them. Right now, the default behaviour is unknown to end users and can be changed later.